xds/internal/balancer/clusterresolver/clusterresolver.go

```diff
@@ -248,15 +248,33 @@ func (b *clusterResolverBalancer) updateChildConfig() {
 		b.logger.Infof("Built child policy config: %s", pretty.ToJSON(childCfg))
 	}
 
-	endpoints := make([]resolver.Endpoint, len(addrs))
-	for i, a := range addrs {
-		endpoints[i].Attributes = a.BalancerAttributes
-		endpoints[i].Addresses = []resolver.Address{a}
+	flattenedAddrs := make([]resolver.Address, len(endpoints))
+	for i := range endpoints {
+		for j := range endpoints[i].Addresses {
```

Comment:
Curious. Are you using this `for` loop syntax over the more common `for idx, entry := range slice` because you want to avoid a copy of the endpoint/address into the `entry` variable? I find the latter more readable, but if you have good reason to write this as you have written it, I wouldn't compel you to change.

Reply:
> Using a for idx, entry := range slice loop will not help as it adds BalancerAttributes to a copy of the address

Makes sense. Thank you.